### PR TITLE
Add more fields to REST API dags/dag_id/details endpoint

### PR DESCRIPTION
**Pull request by ephraimbuddy**
Added more fields to the DAG details endpoint, which is the endpoint for
getting DAG `object` details

### CODE REVIEW

File: airflow/api_connexion/endpoints/dag_endpoint.py [hunk 1]
```diff
@@ -56,6 +56,7 @@ def get_dag(*, dag_id: str, session: Session = NEW_SESSION) -> APIResponse:
 def get_dag_details(*, dag_id: str) -> APIResponse:
     """Get details of DAG."""
     dag: DAG = current_app.dag_bag.get_dag(dag_id)
+
```
Comment (ashb) [inline airflow/api_connexion/endpoints/dag_endpoint.py]:
```suggestion
```

(to avoid touching this file)

**Review by ashb — COMMENTED** [review]
Many of those properties are internal detail of how things are stored/implemented and I'm not sure it make senses to include them in the output.

Specifically:

- edge_info this feels out of place on the DAG details endpoint (and is just an implementation detail of how we store edge labels) but maybe this is the only place it makes sense to include. If we want to keep this here we should specify the type more tightly than just "object"
- partial will never be True on a dag returned by the API. Remove
- has_*_callback is an artefact of serialization. Remove?
- user_defined_* -- what value does this even have when those are set? What could an API consumer do with that info. Remove?
- default_args: is this relevant to anything?
- jinja_environment_kwargs: Remove?
- render_template_as_native_obj: Relevant? Remove?

**Comment by ephraimbuddy** [comment]
> Many of those properties are internal detail of how things are stored/implemented and I'm not sure it make senses to include them in the output.
> 
> Specifically:
> 
> * edge_info this feels out of place on the DAG details endpoint (and is just an implementation detail of how we store edge labels) but maybe this is the only place it makes sense to include. If we want to keep this here we should specify the type more tightly than just "object"
> * partial will never be True on a dag returned by the API. Remove
> * has_*_callback is an artefact of serialization. Remove?
> * user_defined_* -- what value does this even have when those are set? What could an API consumer do with that info. Remove?
> * default_args: is this relevant to anything?
> * jinja_environment_kwargs: Remove?
> * render_template_as_native_obj: Relevant? Remove?

Looks like we don't really need to add more to what is already existing...

**Comment by ephraimbuddy** [comment]
`render_template_as_native_obj` as native object looks relevant.

File: airflow/api_connexion/openapi/v1.yaml [hunk 2]
```diff
@@ -2912,7 +2912,49 @@ components:
...
+                *New in version 2.3.0*.
+            is_paused_upon_creation:
+              type: boolean
+              readOnly: true
+              nullable: true
+              description: |
+                Whether the DAG is paused upon creation.
+
+                *New in version 2.3.0*
+            last_loaded:
```
Comment (ashb) [inline airflow/api_connexion/openapi/v1.yaml]: I wonder if this should be called last_parsed instead?

Comment (ephraimbuddy) [reply airflow/api_connexion/openapi/v1.yaml]: +1. That's more in line with Airflow terms

**Review by ashb — APPROVED** [review]

**Pull request merged**